Fix remaining notarization issues from #8486

poshul opened the pull request:
## Description

<!-- Please include a summary of the change and which issue is fixed here. -->

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.seqan.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **Chores**
  * Enhanced macOS build security with improved code signing and notarization workflows.
  * Strengthened build infrastructure with better keychain management and sensitive data protection in logging.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

poshul committed:
address coderabbot issues from #8486

poshul commented:
@jpfeuffer sometime during the package step, the build hangs. I assume there is some request for interaction from the user, but I'm unfamiliar as to what. Can you opine on the issue?

jpfeuffer commented:
Hmm I think a few minutes might be expected for notarization. Do you have a link?

poshul commented:
Here's it doing it in the nightly last night: https://github.com/OpenMS/OpenMS/actions/runs/20201730812/job/57994091684

poshul commented:
and it's currently waited for ~30 minutes on https://github.com/OpenMS/OpenMS/actions/runs/20208035879/job/58009592384

poshul committed:
Add build.keychain to security list-keychains

Add build.keychain to the list of keychains for code signing.

jpfeuffer commented:
Maybe it's the logo that I included back?? I don't think I changed much else in packaging.
Notarization should be done outside of cmake currently.

poshul commented:
> Maybe it's the logo that I included back?? I don't think I changed much else in packaging. Notarization should be done outside of cmake currently.

The devil-machine suggested that it's an issue with the the keychain not getting added to the default keychain search list. I added a line to explicitly do that, so lets see if it works.

jpfeuffer commented:
Hmm maybe. But I don't think I changed the logic there. I just added a new entry or something.

poshul committed:
Change ctest verbosity level in CI workflow (#8497)

jpfeuffer commented:
CAnnot reproduce locally. Takes two minutes. Also the logo is there.
I do not have a cert locally though. Signing with "-"

poshul commented:
For what its worth, this is the same behaviour that I saw before I was properly importing the cert last year, see https://github.com/OpenMS/OpenMS/pull/7803/changes/066d9ebd0f9ff65fb9706e1c1d694af086dfcd18 which fixed it. No idea why it's behaving this way now again.

jpfeuffer commented:
Feels like CMake and ninja are just too stupid and check every file in the build tree or something when using cpack as a cmake target.
Maybe we should try to call cpack from outside of cmake, too. (i.e. not using the "dist" target, which calls cpack from CMake)
I.e. build the all target with docs and everything. Maybe even the install target?

Then call cpack.

Globbing directories also takes quite some time on my mac. I wonder if the disk on GitHub is just so slow that this gets infinitely slower.

jpfeuffer commented:
I am saying this because the build seems to be stuck on

Re-checking globbed directories...

jpfeuffer commented:
I introduced the dist target to make sure that all the important targets are built before calling cpack and also signed_dist to include the signing into cmake.

I guess these days when everything is on CI, we should just do

ninja install and cpack from outside of cmake.

poshul commented:
> I am saying this because the build seems to be stuck on
> 
> Re-checking globbed directories...

Yeah, but that was what it said when the actual issue was missing certs last time.

jpfeuffer commented:
Ok you can check what I changed in the certs I think I just added a print statement and an additional program (product build) that is allowed to open/use the cert.
And renamed the certs of course

jpfeuffer committed:
Add productsign to key-partition-list in CI workflow

jpfeuffer commented:
@copilot in the package step. Instead of executing everything in the completely obfuscated ctest/cmake script, just reconfigure the project with the given env vars passed as -D flags.
Then run ninja install and cpack

Keep the old step commented out for future use or reversal.

poshul committed:
Update notarize.sh to include ASC_TEAMID parameter

TEAM_ID is now required even with only one team.

poshul committed:
Add teamID to notarize fix producbuild key permissions

jpfeuffer commented:
@poshul the notarization script does not pass team ID to the second call for logs.

poshul committed:
Add team ID to notarization log fetching

poshul commented:
> @poshul the notarization script does not pass team ID to the second call for logs.

Fixed it now. Thats what I get for trying trying to do too many things at once.

jpfeuffer commented:
Hmm I was expecting that when I saw what it installs! We need to sign all the libs (and executables?) from the third parties, too.

Copilot committed:
Initial plan

Copilot committed:
Add macOS code signing for thirdparty components for notarization

Co-authored-by: jpfeuffer <8102638+jpfeuffer@users.noreply.github.com>

Copilot committed:
Add check for THIRDPARTY directory existence before signing

Co-authored-by: jpfeuffer <8102638+jpfeuffer@users.noreply.github.com>

Copilot committed:
Sign each thirdparty component individually for macOS notarization

Co-authored-by: jpfeuffer <8102638+jpfeuffer@users.noreply.github.com>

jpfeuffer committed:
Merge pull request #8506 from OpenMS/copilot/sign-libraries-and-executables

Sign thirdparty components for macOS notarization

poshul committed:
Merge branch 'develop' into fix_notarization
